packages/core/src/core.ts

```diff
@@ -310,7 +320,13 @@ export class McpServer {
       version: options.version,
     };
     this.schemaAdapter = options.schemaAdapter;
-    this.logger = options.logger || console;
+    // Default logger writes all messages to stderr for better visibility
+    this.logger = options.logger || {
+      error: console.error,
+      warn: console.error,
+      info: console.error,
+      debug: console.error,
+    };
```

Comment:
i don't think we should do this. 

either no-op the noisier levels by default or switch back to "console-by-default" 

some reasoning here: `stderr` is semantically treated as errors (by downstream tooling). you don't want to treat debug logs as errors.

Reply:
there is another take that libraries should be quiet by default and enable verbose logging easily, not sure how you feel about that apporach instead